### ISSUE TITLE
Round to nearest grid line for vertex/side movement behavior when moving between major and minor grid

### ISSUE THREAD

**Issue by BLFiedler** [phetsims/quadrilateral#292]
Branching from issue #146:

When using a shift-modifier to move to the "smaller grid", using bigger steps (releasing shift) can put you into a position where you are never able to be on the visual grid again for vertex or side movement.

![image](https://user-images.githubusercontent.com/47331395/208156744-33a384de-48ef-408e-8605-257545982856.png)

Proposing shortening the step when off the grid. Getting back onto the grid intersections requires two key presses, but feels better than jumping to the "nearest" intersection (which may be unintuitive even if algorithmically predictable).

![image](https://user-images.githubusercontent.com/47331395/208157190-a775cff8-7542-4d73-92d6-0e73e0420b6c.png)

In conjunction with #291 , @jessegreenberg suggested some solutions for mouse drag, but said we could implement a default behavior and see how it feels before creating special behavior (given mouse is not as XY restricted).

- [ ] Implement rounding to nearest grid line when pressing non-shift modified directional buttons

**Comment by jessegreenberg** [comment]
OK, done for Vertices, I think they feel OK. Also added to sides, but there are cases where I am not sure how it should behave. For example, take a look at this side: 
![image](https://user-images.githubusercontent.com/6396244/209026437-bd8f215c-12c1-40ed-a0c7-9d78e14b7ac4.png)

What should happen when you press the "up" arrow here.  In master, both Vertices will snap to nearest grid line, changing the side's tilt. Maybe translate both by the smallest delta?

**Comment by jessegreenberg** [comment]
For some reason there is wobble from continuous drag in master, so if "dragging by smallest delta" doesn't feel good we may still want to find a different fix:
![rts2](https://user-images.githubusercontent.com/6396244/209728178-b050405d-eabe-4a60-8b4b-5287c9fe69be.gif)

**Comment by jessegreenberg** [comment]
I found the cause of https://github.com/phetsims/quadrilateral/issues/292#issuecomment-1366221897, so I think we can fix that if we don't like moving each vertex by the smallest delta.

But "moving by the smallest delta" was added above. The motion is consistent, but sometimes feels a bit jumpy. For example, put two vertices one minor grid interval apart and drag the side. It will alternate between moving by 1 and 3 minor grid intervals every step. It may not be a problem and I can't think of a better solution. Over to @BLFiedler to review.
